packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -270,6 +270,14 @@ export function createPatchedFetcher(
                 collectedTags.push(tag)
               }
             }
+
+            // Add tags of the current cache scope to the locally collected tags
+            // for this fetch call.
+            for (const tag of collectedTags) {
+              if (!tags.includes(tag)) {
+                tags.push(tag)
+              }
+            }
```

Comment:
I was curious... is the end result of these lines + [lines 266-272](https://github.com/vercel/next.js/blob/de4f197e2fde19526ee019fbebb068e962b58fc0/packages/next/src/server/lib/patch-fetch.ts#L266-L272) to set both `tags` and `workUnitStore.tags` to their union?  If so, would `tags.length = 0; tags.concat(collectedTags)` work here instead?

Also, would Next.js welcome a PR with that kind of change?  Or are such changes considered too cosmetic in the grand scale of performance?